### PR TITLE
Change ST framework to make it more compatible

### PR DESCRIPTION
**Pull request by tomdee**

### CODE REVIEW

File: calico_containers/tests/st/utils/workload.py [hunk 1]
```diff
@@ -79,12 +79,16 @@ def _get_ping_function(self, ip):
...
-            ping = "ping6"
+        # Default to "ping"
+        ping = "ping"
+
+        try:
+            version = IPAddress(ip).version
+            assert version in [4, 6]
+            if version == 6:
+                ping = "ping6"
+        except BaseException:
```
Comment (robbrockbank) [inline calico_containers/tests/st/utils/workload.py]: why do you need to catch a BaseException here?

File: calico_containers/tests/st/utils/workload.py [hunk 2]
```diff
@@ -79,12 +79,16 @@ def _get_ping_function(self, ip):
...
-        else:  # if version == 6:
-            ping = "ping6"
+        # Default to "ping"
+        ping = "ping"
+
+        try:
+            version = IPAddress(ip).version
+            assert version in [4, 6]
+            if version == 6:
+                ping = "ping6"
```
Comment (robbrockbank) [inline calico_containers/tests/st/utils/workload.py]: `ping = "ping" if version ==4 else "ping6"` might be nicer than defining it and overwriting it

File: calico_containers/tests/st/utils/workload.py (hunk 1 above)
Comment (tomdee) [inline calico_containers/tests/st/utils/workload.py]: Because I want to ping a hostname.

**Comment by robbrockbank** [comment]
Minor comments.  Not sure I prefer this approach of passing in an options string, but understand why it might be useful so LGTM if you have a preference to do it this way,

**Pull request merged**